experiment/migrate_testgrid_tabs.py

```diff
@@ -0,0 +1,197 @@
+    "fejta-bot-periodics.yaml",
+    # Generated security jobs are generated with the same name as kubernetes/kubernetes
+    # presubmits, but we never want to migrate to the generated ones.
+    "generated-security-jobs.yaml",
+]
+
+def main(testgrid_config, prow_dir):
+    with open(testgrid_config, "r") as confFile:
+        config = ruamel.yaml.load(confFile,
+                                  Loader=ruamel.yaml.RoundTripLoader,
+                                  preserve_quotes=True)
+
+    for dashboard in config["dashboards"]:
+        if "dashboard_tab" in dashboard:
+            for dashTab in dashboard["dashboard_tab"][:]:
+                if assert_tab_keys(dashTab):
+                    move_tab(dashboard, dashTab, prow_dir)
+
+    with open(testgrid_config, "w") as confFile:
+        ruamel.yaml.dump(config, confFile, Dumper=ruamel.yaml.RoundTripDumper)
+        confFile.truncate()
+
+
+def move_tab(dashboard, dashTab, prow_dir):
+    """ moveTab determines if the dashboard tab matches a prow job in prow_dir.
+    If it does, it modifies the prow job file to include the necessary annotations
+    It also deletes from the config the dashboard tab, so there are no duplicates.
+    """
+    dashboardName = dashboard["name"]
+    prowJobName = dashTab["test_group_name"]
+    prowJobFileName = find_prow_job(prowJobName, prow_dir)
+    if prowJobFileName == "":
+        return
+
+    # Matching file found; patch and delete
+    print("Patching {0} in {1}".format(prowJobName, prowJobFileName))
+
+    with open(prowJobFileName, "r") as prowJobFile:
+        prowConfig = ruamel.yaml.load(prowJobFile,
+                                      Loader=ruamel.yaml.RoundTripLoader,
+                                      preserve_quotes=True)
+
+    # For each presubmits, postsubmits, periodic:
+    # presubmits -> <any repository> -> [{name: prowjob}]
+    if "presubmits" in prowConfig:
+        for repo, jobs in prowConfig["presubmits"].items():
+            for (i, job) in enumerate(jobs):
+                if prowJobName == job["name"]:
+                    prowConfig["presubmits"][repo][i] = patch_prow_job(job,
+                                                                       dashTab,
+                                                                       dashboardName)
+    # postsubmits -> <any repository> -> [{name: prowjob}]
+    if "postsubmits" in prowConfig:
+        for repo, jobs in prowConfig["postsubmits"].items():
+            for (i, job) in enumerate(jobs):
+                if prowJobName == job["name"]:
+                    prowConfig["postsubmits"][repo][i] = patch_prow_job(job,
+                                                                        dashTab,
+                                                                        dashboardName)
+    # periodics -> [{name: prowjob}]
+    if "periodics" in prowConfig:
+        for i in range(0, len(prowConfig["periodics"])):
```

Comment:
Loop over prowConfig["periodics"].items()?

Reply:
prowConfig["periodics"] is a list, not a map, so I can just reference `jobs in prowConfig["periodics"]`